### ISSUE TITLE
Invalid case of SC1097 (`Unexpected ==. For assignment, use =. For comparison, use [/[[.`)

### ISSUE THREAD

**Issue by AndiDog** [hadolint/hadolint#362]
- [x] This is a bug report
- [ ] This is a feature request
- [ ] I searched existing issues before opening this one

### Expected behavior

Simple script

```
RUN if [ "$USE_SCCACHE" = 1 ]; then echo yay; fi
```

does not produce any lint warnings.

### Actual behavior

SC1097 is raised

### Steps to reproduce the behavior

```
$ cat Dockerfile
FROM alpine:3

ARG USE_SCCACHE=

RUN if [ "$USE_SCCACHE" = 1 ]; then echo yay; fi
$ hadolint Dockerfile
Dockerfile:5 SC1097 Unexpected ==. For assignment, use =. For comparison, use [/[[.
$ cat scripts.sh
#!/bin/sh
if [ "$USE_SCCACHE" = 1 ]; then echo yay; fi
$ shellcheck scripts.sh
$ hadolint --version
Haskell Dockerfile Linter 1.17.2-no-git
```

**Additional environment details (OS, stack version, etc.)**

macOS

**Comment by lorenzo** [comment]
Apparently, the bug is triggered because of `ARG USE_SCCACHE=` if you make it `ARG USE_SCCACHE=""`  or `ARG USE_SCCACHE` the bug is gone.

I'll investigate, but you should use `ARG USE_SCCACHE` instead for the time being.
